pymie/mie.py

```diff
@@ -303,6 +437,10 @@ def _pis_and_taus(nstop, thetas):
     ang_shape = list(thetas.shape)
     
     # flatten to make calculations easier
+    if isinstance(thetas, Quantity):
```

Comment:
why are the nested "if" statements needed here?

Reply:
because there were cases where units were added to thetas twice, so it was still a quantity after taking the first magnitude.

Reply:
That's a bug, so maybe we don't nest the if statements and correct the places where units are being added twice?

Reply:
I removed the nested if statement and the tests passed. I think I had added the nested if statement in an early stage before I completely realized how the new pint and numpy were handling the units. e.g. the fact that the integrating functions on numpy now preserve the pint units.